hledger-lib/Hledger/Data/Types.hs

```diff
@@ -198,7 +198,7 @@ data Posting = Posting {
       pcomment          :: Text,              -- ^ this posting's comment lines, as a single non-indented multi-line string
       ptype             :: PostingType,
       ptags             :: [Tag],             -- ^ tag names and values, extracted from the comment
-      pbalanceassertion :: Maybe MixedAmount, -- ^ optional: the expected balance in the account after this posting
+      pbalanceassertion :: Maybe Amount, -- ^ optional: the expected balance in the account after this posting
```

Comment:
IIRC the doc should be changed to "the expected balance in this commodity in the account after this posting". I can fix it later.